data/input_files/tests/nonlinear_solid/dyn_amgx_solve.lua

```diff
@@ -47,12 +50,31 @@ nonlinear_solid = {
     mu = 0.25,
     K  = 5.0,
 
+    -- initial conditions
+    initial_displacement = {
+        vec_coef = function (x, y, z)
+            return 0, 0, 0
+        end  
+    },
+
+    initial_velocity = {
+        vec_coef = function (x, y, z)
+            s = 0.1 / 64
+            first = -s * x * x
+            last = s * x * x * (8.0 - x)
+            -- FIXME: How can we detect the dimension?
+            return first, 0, last
```

Comment:
in 2D, we typically keep 3 variables but leave the last one 0.0 (at least in older codes).

Reply:
Yep, the issue is that the current interface does not provide enough information to determine the dimension, so I've assumed 3D for the time being.  A future Axom PR might add the vector type as a first-class interface whose size can be queried.